src/main/java/piano/instruments/InstrumentsPanel.java

```diff
@@ -11,9 +11,13 @@ public InstrumentsPanel(MidiChannel midiChannel) {
         setLayout(new FlowLayout());
         setSize(MainFrameInterface.KEYBOARD_WIDTH, MainFrameInterface.INSTRUMENTS_PANEL_HEIGHT);
         setBackground(Color.BLACK);
-        for (InstrumentButton instrumentButton: new InstrumentButtonsArray()) {
-            instrumentButton.addActionListener(e -> midiChannel.programChange(instrumentButton.getInstrument()));
-            add(instrumentButton);
+        JComboBox combo = new JComboBox();
+        InstrumentDictionary instrumentDictionary= new InstrumentDictionary();
+
+        for (String next: instrumentDictionary.getInstrumentNames()) {
+            combo.addItem(next);
         }
+        combo.addActionListener(e -> midiChannel.programChange(instrumentDictionary.getSound((String) combo.getSelectedItem())));
```

Comment:
Try to find a way to do this without the Map lookup. 
Hint: you can put things other than Strings in the JComboBox